OPUNetGameSelectWnd.cpp

```diff
@@ -652,7 +652,7 @@ void OPUNetGameSelectWnd::OnReceive(Packet &packet)
 		}
 		if (bTwoExternal)
 		{
-			strncat(text, "\nWarning: Address and Port-Dependent Mapping detected\nYou may have difficulty joining games.", sizeof(text));
+			strncat_s(text, "\nWarning: Address and Port-Dependent Mapping detected\nYou may have difficulty joining games.", sizeof(text));

```

Comment:
They are a drop in replacement, except when they're not. Mingw gives a compile error for this one line change. Error:
```
OPUNetGameSelectWnd.cpp:656:130: error: too few arguments to function ‘errno_t strncat_s(char*, size_t, const char*, size_t)’
```

Reference:
https://en.cppreference.com/w/c/string/byte/strncat
It seems to be selecting:
```cpp
errno_t strncat_s(char *restrict dest, rsize_t destsz, const char *restrict src, rsize_t count);
```
Not too sure why the later version of option 1 wasn't chosen:
```cpp
char *strncat( char *restrict dest, const char *restrict src, size_t count );
```

There seems to be a C++ workaround:
https://en.cppreference.com/w/cpp/string/byte/strncat
```cpp
char *strncat( char *dest, const char *src, std::size_t count );
```
Though this seems to bring back the warning with MSVC.

----

The MSDN has:
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/strncat-s-strncat-s-l-wcsncat-s-wcsncat-s-l-mbsncat-s-mbsncat-s-l?view=vs-2017

The relevant overload of the method being:
```cpp
template <size_t size>
errno_t strncat_s(
   char (&strDest)[size],
   const char *strSource,
   size_t count
); // C++ only
```

Reply:
Oh, I see. The first overload doesn't have the "_s". That's why it's not being selected, the name is different. It's the old method that produces the warning. The second overload has the "_s", but it also has the extra `size_t` parameter for the size of the destination buffer.

MSVC has a non-standard extension for C++, where it detects the size of static arrays, which the `text` variable is. Hence why MSVC can handle that one line, but standard C++ can't.